package-build.el

```diff
@@ -1123,12 +1125,12 @@ Returns the archive entry for the package."
      ((not version)
       (error "Unable to check out repository for %s" package-name))
      ((= 1 (length files))
-      (pb/build-single-file-package package-name version (caar files) source-dir target-dir))
+      (pb/build-single-file-package package-name version (caar files) source-dir target-dir dependencies))
      ((< 1 (length  files))
-      (pb/build-multi-file-package package-name version files source-dir target-dir))
+      (pb/build-multi-file-package package-name version files source-dir target-dir dependencies))
      (t (error "Unable to find files matching recipe patterns")))))
 
-(defun pb/build-single-file-package (package-name version file source-dir target-dir)
+(defun pb/build-single-file-package (package-name version file source-dir target-dir dependencies)
```

Comment:
Personally I think we have to create a separate function which returns the dependencies in the `package-build` fashion. Then replace dependencies if the `dependencies` argument passed to `package-build-package`. 

This way user can manually fetch calculated dependencies and decide what to do with it. Merge or replace, then pass the resulted dependencies to `package-build-package`.

This way we won't need to write code for tools like **Cask** ( @milkypostman solution ), and we will have much clear API.